src/pkg/utils/oras.go

```diff
@@ -92,27 +179,247 @@ func (o *OrasRemote) withAuthClient(ref registry.Reference) (*auth.Client, error
+//
+// It also retrieves the descriptors for all image layers that are required by the components.
+//
+// It also respects the `required` flag on components, and will retrieve all necessary layers for required components.
+func (o *OrasRemote) LayersFromRequestedComponents(requestedComponents []string) (layers []ocispec.Descriptor, err error) {
+	root, err := o.FetchRoot()
+	if err != nil {
+		return nil, err
+	}
+
+	pkg, err := o.FetchZarfYAML(root)
+	if err != nil {
+		return nil, err
+	}
+	images := []string{}
+	tarballFormat := "%s.tar"
+	for _, name := range requestedComponents {
+		component := Find(pkg.Components, func(component types.ZarfComponent) bool {
+			return component.Name == name
+		})
+		if component.Name == "" {
+			return nil, fmt.Errorf("component %s does not exist in this package", name)
+		}
+	}
+	for _, component := range pkg.Components {
+		// If we requested this component, or it is required, we need to pull its images and tarball
+		if SliceContains(requestedComponents, component.Name) || component.Required {
+			images = append(images, component.Images...)
+			layers = append(layers, root.Locate(filepath.Join(zarfconfig.ZarfComponentsDir, fmt.Sprintf(tarballFormat, component.Name))))
+		}
+	}
+	if len(images) > 0 {
+		// Add the image index and the oci-layout layers
+		layers = append(layers, root.Locate(root.indexPath), root.Locate(root.ociLayoutPath))
+		// Append the sbom.tar layer if it exists
+		sbomDescriptor := root.Locate(zarfconfig.ZarfSBOMTar)
+		if sbomDescriptor.Digest != "" {
+			layers = append(layers, sbomDescriptor)
+		}
+		index, err := o.FetchImagesIndex(root)
+		if err != nil {
+			return nil, err
+		}
+		for _, image := range images {
+			manifestDescriptor := Find(index.Manifests, func(layer ocispec.Descriptor) bool {
+				return layer.Annotations[ocispec.AnnotationBaseImageName] == image
+			})
+			manifest, err := o.FetchManifest(manifestDescriptor)
+			if err != nil {
+				return nil, err
+			}
+			// Add the manifest and the manifest config layers
+			layers = append(layers, root.Locate(filepath.Join(root.imagesBlobsDir, manifestDescriptor.Digest.Encoded())))
+			layers = append(layers, root.Locate(filepath.Join(root.imagesBlobsDir, manifest.Config.Digest.Encoded())))
+
+			// Add all the layers from the manifest
+			for _, layer := range manifest.Layers {
+				layerPath := filepath.Join(root.imagesBlobsDir, layer.Digest.Encoded())
+				layers = append(layers, root.Locate(layerPath))
+			}
+		}
+	}
+	message.Infof("Pulling %+v", layers)
+	return layers, nil
+}
+
+// PullPackage pulls the package from the remote repository and saves it to the given path.
+//
+// layersToPull is an optional parameter that allows the caller to specify which layers to pull.
+//
+// The following layers will ALWAYS be pulled if they exist:
+//   - zarf.yaml
+//   - checksums.txt
+//   - zarf.yaml.sig
+func (o *OrasRemote) PullPackage(out string, concurrency int, layersToPull ...ocispec.Descriptor) error {
```

Comment:
Can we make `out` more obvious that it's a directory path and not a singular output binary of the package <3